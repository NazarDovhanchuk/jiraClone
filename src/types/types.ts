import { AppState } from '../store/storeTypes';

export type SelectFromState<T> = (state: AppState) => T;
