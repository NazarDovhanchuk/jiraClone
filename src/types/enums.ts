export enum STATUSES {
  TODO = 'TODO',
  IN_PROGRESS = 'IN_PROGRESS',
  DONE = 'DONE',
}
