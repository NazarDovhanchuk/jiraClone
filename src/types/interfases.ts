import { STATUSES } from './enums';

export interface Todo {
  userId: number,
  id: number,
  title: string,
  status: STATUSES,
}
