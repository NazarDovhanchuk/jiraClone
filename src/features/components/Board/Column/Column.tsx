import React from 'react';
import TicketItem from '../../TicketItem/TicketItem';
import { STATUSES } from '../../../../types/enums';
import { Todo } from '../../../../types/interfases';

interface ColumnProps {
  title: string;
  status: STATUSES;
  items: Todo[];
}

const Column: React.FC<ColumnProps> = ({ title, status, items }): JSX.Element => {
  const filterTodo = items.filter((todo) => todo.status === status);

  return (
    <div>
      <h3 style={{ textAlign: 'center' }}>{title}</h3>
      <div className="progress">
        {filterTodo.map((el) => (
          <TicketItem key={el.id} title={el.title} status={el.status} id={el.id} />
        ))}
      </div>
    </div>
  );
};

export default Column;
