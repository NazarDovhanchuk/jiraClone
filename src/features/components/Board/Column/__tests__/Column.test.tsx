import React from "react";
import rendered from "react-test-renderer";
import { cleanup, render } from "@testing-library/react";

import Column from "../Column";
import { STATUSES } from "../../../../../types/enums";

afterEach(cleanup);

describe("Column", () => {
  it("column with render", () => {
    const columnComponent = rendered.create(<Column title={""} status={STATUSES.TODO} items={[]} />);
    const tree = columnComponent.toJSON();

    expect(tree).toMatchSnapshot();
  });
});