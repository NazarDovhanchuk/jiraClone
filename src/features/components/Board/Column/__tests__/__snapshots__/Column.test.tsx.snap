// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button button without text 1`] = `
<div>
  <h3
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    
  </h3>
  <div
    className="progress"
  />
</div>
`;

exports[`Column button without text 1`] = `
<div>
  <h3
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    
  </h3>
  <div
    className="progress"
  />
</div>
`;

exports[`Column column with  1`] = `
<div>
  <h3
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    
  </h3>
  <div
    className="progress"
  />
</div>
`;

exports[`Column column with render 1`] = `
<div>
  <h3
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    
  </h3>
  <div
    className="progress"
  />
</div>
`;
